In [1]:
import netket as nk

In [56]:
L =4 
g = nk.graph.Chain(length=L,pbc=False)

In [57]:
# Define the Hilbert space based on this graph
# We impose to have a fixed total magnetization of zero
hi = nk.hilbert.Spin(s=0.5, N=g.n_nodes)

In [58]:
# calling the Heisenberg Hamiltonian
#ha = nk.operator.Heisenberg(hilbert=hi, graph=g)
ha = nk.operator.Ising(hilbert=hi, graph=g, h=-1.0) # -h\sum_i \sigma_i^{(x)} +J\sum_{\langle i,j\rangle} \sigma_i^{(z)}\sigma_j^{(z)}`.

### ED

In [59]:
# compute the ground-state energy (here we only need the lowest energy, and do not need the eigenstate)
evals = nk.exact.lanczos_ed(ha, compute_eigenvectors=False, k=2)
print(evals/L)
exact_gs_energy = evals[0]/L
print("The exact ground-state energy is E0=", exact_gs_energy)

[-1.18969262 -1.01604444]
The exact ground-state energy is E0= -1.1896926207859084


### ED results Heisenberg:
- nq = 3: -1.333333333333334
- nq = 4: -1.6160254037844388
- nq = 5: -1.5423090026543964

### ED results TFIM:
- nq = 3: -1.1646530691449783 (first excited state: -0.86795849)
- nq = 4: -1.1896926207859086 (first excited: -1.01604444)
- nq = 5: -1.2053348366664525 (first excited: -1.09148297)
- nq = 6: -1.21603830175979 (first exc: -1.13568051)
- nq = 7: -1.2238246  (first excited: -1.16409405)
- nq = 8: -1.22974393 (first excited: -1.18360975)